CLOUDSTACK-8968: UI icon over VM snapshot to deploy user instance

nitin-maharana opened the pull request:
Added a new Icon in Instance page to launch the VM from the snapshot.

A new icon over VM snapshot object, which upon invoked will open a deploy user instance wizard, where there is no choice to choose zone and ISO or template to deploy the user instance from.

1) A new icon to indicate deploy user instance from VM snapshot
2) The new icon is placed as an operation over each VM snapshot object.
3) Clicking this icon will show the wizard to deploy user instance, which is same as wizard for normal user instance except that zone selection page and ISO/template selection page would not be present in this wizard.

DaanHoogland commented on ui/scripts/instanceWizard.js:
```diff
@@ -731,25 +746,33 @@
-                }
-            }
+//            Currently it is not supporting in backend
+//            if (args.$wizard.find('input[name=rootDiskSize]').parent().css('display') != 'none')  {
```
please do not leave code commented in. this block should be deleted.

bhaisaab commented:
@nitin-maharana if you can address what Daan has commented today, along with fix tabs with spaces (if any) I can help review this and merge this; before 4.7.0/master freeze is announced today.

remibergsma commented:
@nitin-maharana @bhaisaab I looked around but just don't find what's changed. Please post screenshots of what is changed or instruct exactly how it should work. Thanks!

remibergsma commented:
Hmm.. this isn't supported on KVM, is it? That may be why I can't find it.

nitin-maharana commented:
@bhaisaab @DaanHoogland I updated the change with current 4.6 branch and removed the commented codes. @remibergsma This only supports in VMware. Thanks :)

DaanHoogland commented:
@nitin-maharana @bhaisaab both I and @remibergsma don't work with vmware as part of our cloudstack installs, can you do testing and preferably write tests for proving the continued funtioning of the change?

DaanHoogland commented:
@bhaisaab did you test this? should we merge before the RC tonight?

bhaisaab commented:
@DaanHoogland sorry missed your message, I did not test this

nitin-maharana commented:
Step-1:
=====
![pr_1150_ss1_nitin](https://cloud.githubusercontent.com/assets/12583725/11789386/1d14793c-a2bc-11e5-9184-f16b84af91f4.png)

Step-2:
=====
Skipped 1st and 2nd steps

![pr_1150_ss2_nitin](https://cloud.githubusercontent.com/assets/12583725/11789434/636162f6-a2bc-11e5-8746-5fd2411803df.png)

Step-3:
=====
Last step

![pr_1150_ss3_nitin](https://cloud.githubusercontent.com/assets/12583725/11789449/7404f2ee-a2bc-11e5-8bc7-b196a1bae76d.png)

nitin-maharana commented:
cc @bhaisaab

bhaisaab commented:
@nitin-maharana sorry don't have the infra to setup vmware and test right now, so cannot help test it

nitin-maharana commented:
@remibergsma : Any updates on this? Can you please look at the change if you have a setup. Thanks.

remibergsma commented:
Sorry I have no vmware setup to test this.

nitin-maharana commented:
Closing this PR as made a new PR #1320 (Against 4.7 which will be merged in master later).
